proof/refine/ARM/Syscall_R.thy

```diff
@@ -1156,98 +1182,121 @@ crunch valid_duplicates'[wp]: rescheduleRequired "\<lambda>s. vs_valid_duplicate
 
 crunch valid_duplicates'[wp]: setThreadState "\<lambda>s. vs_valid_duplicates' (ksPSpace s)"
 
+crunches reply_from_kernel
+  for pspace_aligned[wp]: pspace_aligned
+  and pspace_distinct[wp]: pspace_distinct
+  and valid_objs[wp]: valid_objs
+  (simp: crunch_simps wp: crunch_wps)
+
+crunches replyFromKernel
+  for valid_objs'[wp]: valid_objs'
+  and valid_release_queue[wp]: valid_release_queue
+  and valid_release_queue'[wp]: valid_release_queue'
+  (simp: crunch_simps wp: crunch_wps)
+
+(* Note: the more elaborate preconditions on the abstract side come mostly from pinv_corres *)
```

Comment:
By more elaborate do you mean in comparison to the previous preconditions? If so I would rewrite it to something like
```
Note: the preconditions on the abstract side are based on those of pinv_corres
```
We should be careful with comments referring to previous (and even sometimes existing) states of the repository, since they are very likely to confuse people coming across them in the future.

Separately, what are you trying to convey to future maintainers who come across this comment, and can you make it more explicit? My guess is something along the lines of 'if this proof breaks, check if any new preconditions have been added to `pinv_corres`'.

Reply:
Ah, by "more elaborate" I meant "not just `invs` or `valid_sched`". I'll make it clearer.